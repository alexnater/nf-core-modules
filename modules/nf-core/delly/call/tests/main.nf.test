nextflow_process {

    name "Test Process DELLY_CALL"
    script "../main.nf"
    process "DELLY_CALL"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "delly"
    tag "delly/call"

    test("germline - bcf") {
        when {
            params {
                suffix = 'bcf'
            }
            process {
                """
                input[0] = [
                    [ id:'control' ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                    [],
                    [],
                    [],
                    [],
                    []
                    ]
                input[1] = [
                    [ id:'human' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }
    }

    test("germline - vcf") {
        when {
            params {
                suffix = 'vcf'
            }
            
            process {
                """
                input[0] = [
                    [ id:'control' ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                    [],
                    [],
                    [],
                    [],
                    []
                    ]
                input[1] = [
                    [ id:'human' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }
    }

    test("germline - genotyping") {
        setup {
            run("DELLY_CALL", alias: "DELLY_PRECALL") {
                script "../main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'control' ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                        [],
                        [],
                        [],
                        [],
                        []
                        ]
                    input[1] = [
                        [ id:'human' ],
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                        ]
                    """
                }
            }
        }

        when {
            params {
                suffix = 'bcf'
            }
            process {
                """
                ch_input = Channel.of([
                    [ id:'control' ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true)
                ]).combine(DELLY_PRECALL.out.bcf)
                  .combine(DELLY_PRECALL.out.csi)
                  .map { meta, bam, bai, meta2, bcf, meta3, csi -> [ meta, bam, bai, [], [], bcf, csi, [] ] }
                input[0] = ch_input
                input[1] = [
                    [ id:'human' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }
    }

    test("germline - stub") {

        options "-stub"

        when {
            params {
                suffix = 'bcf'
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [],
                    [],
                    [],
                    [],
                    [],
                    [],
                    []
                    ]
                input[1] = [
                    [:],
                    [],
                    []
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }

    }

    test("somatic - bcf") {
        when {
            params {
                suffix = 'bcf'
            }
            process {
                """
                input[0] = [
                    [ id:'tumor_control' ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                    [],
                    [],
                    []
                    ]
                input[1] = [
                    [ id:'human' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }
    }

    test("somatic - vcf") {
        when {
            params {
                suffix = 'vcf'
            }
            
            process {
                """
                input[0] = [
                    [ id:'tumor_control' ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                    [],
                    [],
                    []
                    ]
                input[1] = [
                    [ id:'human' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }
    }
}
