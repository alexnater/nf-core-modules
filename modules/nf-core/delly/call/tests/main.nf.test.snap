{
    "germline - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "control"
                        },
                        "control.vcf.gz:md5,5fd37a553800360383531e2e94bb34f7"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "control"
                        },
                        "control.vcf.gz.tbi:md5,4cb176febbc8c26d717a6c6e67b9c905"
                    ]
                ],
                "2": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ],
                "bcf": [
                    [
                        {
                            "id": "control"
                        },
                        "control.vcf.gz:md5,5fd37a553800360383531e2e94bb34f7"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "control"
                        },
                        "control.vcf.gz.tbi:md5,4cb176febbc8c26d717a6c6e67b9c905"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T10:01:53.508954374"
    },
    "somatic - bcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.bcf:md5,459e93a1320c034640f1186145b40b9b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.bcf.csi:md5,66545c77fc791a02dadc7fc252d04635"
                    ]
                ],
                "2": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ],
                "bcf": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.bcf:md5,459e93a1320c034640f1186145b40b9b"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.bcf.csi:md5,66545c77fc791a02dadc7fc252d04635"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T10:02:11.697572328"
    },
    "somatic - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.vcf.gz:md5,f8b7f81ab296ef51447db984db7f28c9"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.vcf.gz.tbi:md5,4cb176febbc8c26d717a6c6e67b9c905"
                    ]
                ],
                "2": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ],
                "bcf": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.vcf.gz:md5,f8b7f81ab296ef51447db984db7f28c9"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "tumor_control"
                        },
                        "tumor_control.vcf.gz.tbi:md5,4cb176febbc8c26d717a6c6e67b9c905"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T10:02:21.901906278"
    },
    "germline - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ],
                "bcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-21T16:54:36.024035492"
    },
    "germline - bcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "control"
                        },
                        "control.bcf:md5,cae40c6bf83e68e387b275b258c10737"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "control"
                        },
                        "control.bcf.csi:md5,66545c77fc791a02dadc7fc252d04635"
                    ]
                ],
                "2": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ],
                "bcf": [
                    [
                        {
                            "id": "control"
                        },
                        "control.bcf:md5,cae40c6bf83e68e387b275b258c10737"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "control"
                        },
                        "control.bcf.csi:md5,66545c77fc791a02dadc7fc252d04635"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cf72d737ee881a7cd7eab87abf93b94c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T10:01:44.572584694"
    }
}